mcs/class/corlib/System/TimeZoneInfo.cs

```diff
@@ -116,11 +116,11 @@ static TimeZoneInfo CreateLocal ()
 
 			try {
 				return FindSystemTimeZoneByFileName ("Local", "/etc/localtime");	
-			} catch {
+			} catch (TimeZoneNotFoundException) {
```

Comment:
Let A be `/etc/localtime` and B be `/usr/share/zoneinfo/localtime`.
When A is valid we return without checking B validity.
Symmetrically when B is valid we would also return it without checking A validity.

I don't think any of the above options is good. There is no reason to enforce A over B or B over A.

The logic I am following is that there is no reason to fail just because we decided to enforce A over B while if we would choose to enforce B over A it would work fine.